[Team-02][FE] 1주차 첫 번째 PR

Serin-Kim opened the pull request:
안녕하세요! 메토리입니다 🚙 =3=3=3

## **🤖 데모**

[데모 링크 (つ✧.✧)つ-✧҉*](https://philsogooood.github.io/sidedish/)

![ezgif com-gif-maker (12)](https://user-images.githubusercontent.com/68533016/164242652-26cd94c2-e477-44af-bd9c-f1c54163a6b2.gif)

 
## **✨ 구현 사항**
### **1. Header Component 구현**

- [x] html 마크업
- [x] css 구현 
    - 헤더 css  
    - 서브카테고리에 hover 이벤트가 발생하면 텍스트 색이 변경되고 밑줄이 생긴다.
- [x] 레이아웃 및 메뉴 노출 효과
     - 메뉴 영역에 hover 이벤트가 발생하면 서브카테고리가 나타난다.
     - 마우스가 메뉴 영역을 벗어나면 서브카테고리 레이어가 사라진다. 


### **2. specialPromotion Component 구현**
- [x] Theme 영역 구현
- [x] Tab 영역 구현 
    - 클릭 이벤트가 발생하면 해당 탭의 태그에 `Selected` 클래스가 추가된다. 
    - `Selected` 클래스가 추가되면 선택된 탭 하단에 line이 렌더링 된다. 
- [x] bestSideDishContainer 영역 구현 
    - 선택된 탭 영역에 해당하는 반찬이 렌더링 된다. (임시데이터 렌더링) 

### **3. goodsBlock Component 구현**
- [x] goodsBlock Component 구현
    - 재사용되는 각 메뉴들의 컴포넌트를 구현한다. 

## **🤔 이후 계획**
-  슬라이드 컴포넌트 구현 
- 메뉴 카드 hover 효과 (새벽 배송 / 전국 택배 라벨)
- 목서버 연결 (json 서버)
- 리액트와 친해지기 (useEffect ...)

PhilSoGooood committed:
docs: FE, BE README update

mogooee committed:
Header Component 구현 (#4)

* docs: update README

* rename: 디렉토리명 변경

각 클래스별 디렉토리명을 풀네임으로 변경하였습니다

* Update issue templates

* docs: README 수정

브랜치 전략 및 커밋룰 수정

* feat: header HTML markup

header Component HTML 마크업

* feat: 레이아웃 및 메뉴 노출 효과
메뉴 영역에 mouseover, mosueleave 이벤트 발생에 따라 서브카테고리 레이어 노출,숨김

* design: 서브카테고리 hover 구현

서브카테고리에 hover가 발생하면 텍스트에 밑줄 생성, 색상 변경

* design: header CSS 구현

메뉴 영역에 mouseover 이벤트 발생시 동적으로 height 조정

* refactor: 카테고리리스트 렌더링 구조 변경

mainCategory, subCategory 분리하여 렌더링 -> categoryList로 한 번에 렌더링

Co-authored-by: rkolx <olx.rko.o@gmail.com>
Co-authored-by: PhilSoGooood <peelhw@gmail.com>
Co-authored-by: Sangpil Hwang <87455844+PhilSoGooood@users.noreply.github.com>

Serin-Kim committed:
specialPromotion, goodsBlock Component 구현 (#6)

* docs: update README

* rename: 디렉토리명 변경

각 클래스별 디렉토리명을 풀네임으로 변경하였습니다

* Update issue templates

* docs: README 수정

브랜치 전략 및 커밋룰 수정

* feat: Theme 영역 구현

Theme 영역 html, css 마크업

* feat: goodsBlock Component 구현

목데이터를 사용해서 재사용되는 각 메뉴들의 컴포넌트를 구현했다.

* style: prettier 적용

.prettierrc 파일을 생성해 적용했다.

* feat: bestSideDishContainer 영역 구현

선택된 탭 영역에 해당하는 반찬이 렌더링 된다.

* feat:  selected Tab 렌더링

선택된 탭 하단에 line이 렌더링된다.

* refactor: goodsLabel 태그 변수화

* design: specialPromition, goodsBlock Component CSS 구현

Co-authored-by: rkolx <olx.rko.o@gmail.com>
Co-authored-by: PhilSoGooood <peelhw@gmail.com>
Co-authored-by: Sangpil Hwang <87455844+PhilSoGooood@users.noreply.github.com>

snowjang24 commented on frontend/src/js/components/SpecialPromotion.js:
```diff
@@ -0,0 +1,52 @@
+          goodsName={element.name}
+          goodsDescription={element.description}
+          goodsPrice={element.price}
+          goodsLabel={element.label}
```
모든 이름에 goods가 붙을 필요는 없을 것 같습니다:)

snowjang24 commented on frontend/src/js/components/SpecialPromotion.js:
```diff
@@ -0,0 +1,52 @@
+    </li>
+  ));
+
+  const bestSideDish = goodsData
```
이 부분을 따로 함수 컴포넌트로 분리하는 것은 어떨까요 ?

snowjang24 commented on frontend/src/js/components/SpecialPromotion.js:
```diff
@@ -0,0 +1,52 @@
+      key={element.id}
+      onClick={() => {
+        setTabState(element.tab.title);
+      }}
```
handler함수를 따로 분리하는 것이 좋을 것 같습니다. :)

snowjang24 commented on frontend/src/js/components/SpecialPromotion.js:
```diff
@@ -0,0 +1,52 @@
+import '../../css/SpecialPromotion.css';
+
+function SpecialPromotion() {
+  const [tabState, setTabState] = useState('풍부한 고기 반찬');
```
텍스트를 매핑하는 Object를 따로 두어 분리하면 재사용할 때 용이합니다 😃

mogooee committed:
header, special Promotion, goodsBlock Component CSS 수정 (#8)

* docs: update README

* rename: 디렉토리명 변경

각 클래스별 디렉토리명을 풀네임으로 변경하였습니다

* Update issue templates

* docs: README 수정

브랜치 전략 및 커밋룰 수정

* feat: Theme 영역 구현

Theme 영역 html, css 마크업

* feat: goodsBlock Component 구현

목데이터를 사용해서 재사용되는 각 메뉴들의 컴포넌트를 구현했다.

* style: prettier 적용

.prettierrc 파일을 생성해 적용했다.

* feat: bestSideDishContainer 영역 구현

선택된 탭 영역에 해당하는 반찬이 렌더링 된다.

* feat:  selected Tab 렌더링

선택된 탭 하단에 line이 렌더링된다.

* refactor: goodsLabel 태그 변수화

* design: specialPromition, goodsBlock Component CSS 구현

* design:  header, special Promotion, goodsBlock Component CSS 수정

* chore: package.json githubPages 배포 설정

Co-authored-by: rkolx <olx.rko.o@gmail.com>
Co-authored-by: PhilSoGooood <peelhw@gmail.com>
Co-authored-by: Sangpil Hwang <87455844+PhilSoGooood@users.noreply.github.com>
Co-authored-by: Serin-Kim <serin9864@gmail.com>

snowjang24 commented on frontend/src/js/components/GoodsBlock.js:
```diff
@@ -0,0 +1,38 @@
+import {eventLabelIcon, launchingLabelIcon} from '../constants/iconPath.js';
+import '../../css/GoodsBlock.css';
+
+function GoodsBlock(props) {
```
destructuring 을 활용하여 어떤 props들이 전달되는지 보여주는 것도 좋습니다 👍

snowjang24 commented on frontend/src/js/components/Header.js:
```diff
@@ -0,0 +1,45 @@
+        </a>
+        <a href="#!" className="myCart">
+          <img className="myCartIcon" src={myCartIcon} alt="myCartIcon"></img>
+        </a>
```
링크를 하나의 컴포넌트로 만들어서 순회를 통해 뿌려주는 것도 좋을 것 같습니다 : )

snowjang24 commented on frontend/src/js/components/Header.js:
```diff
@@ -0,0 +1,47 @@
+        <nav className="navBar">
+          <a href="#!" className="search">
+            <img className="searchIcon" src={searchIcon} alt="searchIcon"></img>
+          </a>
```
링크를 하나의 컴포넌트로 분리할 수있을 것 같습니다 :)

snowjang24 reviewed (commented):
안녕하세요, 이번에 리뷰를 담당하게 된 스노우☃️입니다.

전체적으로 깔끔하게 로직을 잘 작성하셨네요 :) 

컴포넌트나 핸들러 함수를 좀 더 분리하면 깔끔하게 보이고 좋을 것 같습니다 👍 

아직은 초반이라 많은 코멘트를 달아 놓지 못했지만, 앞으로의 코드들이 기대 됩니다 :)
